### PR TITLE
Peer.downloadDependencies: build GetDataMessage in private method

### PR DESCRIPTION
**Pull request by msgilligan**
Construct `getdata` a little more functionally/atomically.

Unfortunately, depends on PR #2350, so will need a rebase once that is merged.

### CODE REVIEW

**Comment by msgilligan** [comment]
@schildbach rebased and ready for review.

**Comment by msgilligan** [comment]
@schildbach I just did another gratuitous rebase on this one. I should probably stop doing this, but it simplifies the graphical view of my outstanding branches locally.

File: core/src/main/java/org/bitcoinj/core/Peer.java [hunk 1]
```diff
@@ -870,6 +869,18 @@ protected void processTransaction(final Transaction tx) throws VerificationExcep
...
+    /**
+     * Build a GetDataMessage to query multiple transactions by ID
+     * @param txIds A set of transaction IDs to query
+     * @return A GetDataMessage that will query those IDs
+     */
+    private GetDataMessage buildMultiTransactionDataMessage(Set<Sha256Hash> txIds) {
+        GetDataMessage getdata = new GetDataMessage(params);
+        txIds.forEach(txId ->
+            getdata.addTransaction(txId, vPeerVersionMessage.isWitnessSupported()));
+            return getdata;
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Peer.java:881]: Is this indentation correct? Seems like it should be one indent less.

**Comment by schildbach** [comment]
Looks good, apart from the one comment.

File: core/src/main/java/org/bitcoinj/core/Peer.java (hunk 1 above)
Comment (msgilligan) [reply core/src/main/java/org/bitcoinj/core/Peer.java]: Fixed and force-pushed.

**Comment by msgilligan** [comment]
@schildbach Indentation fixed, change force-pushed, and ready to merge!

**Comment by schildbach** [comment]
Merged.

**Pull request closed without merging**